tests/components/nextcloud/test_sensor.py

```diff
@@ -0,0 +1,89 @@
+"""The tests for the nextcloud sensor component."""
+
+from unittest import TestCase
+
+from homeassistant.components.nextcloud.sensor import NextcloudSensor
+
+from tests.common import get_test_home_assistant, setup_component
+
+API_RESPONSE = {"nextcloud_system_version": "17.0.3.1"}
+
+VALID_CONFIG = {
+    "sensor": {
+        "platform": "nextcloud",
+        "url": "https://nextcloud.sample.org",
+        "username": "username",
+        "password": "password",
+    }
+}
+
+INVALID_CONFIG = {
+    "sensor": {
+        "platform": "nextcloud",
+        "url": [],  # Url as empty list to invalidate configuration
+        "username": "username",
+        "password": "password",
+    }
+}
+
+
+class testSensorSetup(TestCase):
```

Comment:
All new tests must be standalone pytest test functions.

See eg:
https://github.com/home-assistant/core/blob/0a25e86fab78caeb28be6162255f5e3561527687/tests/components/manual/test_alarm_control_panel.py#L85-L121